src/apm/java.go

```diff
@@ -55,28 +87,35 @@ func InjectJavaagent(javaSpec v1alpha1.Java, pod corev1.Pod, index int) (corev1.
 		container.Env[idx].Value = container.Env[idx].Value + javaJVMArgument
 	}
 
-	container.VolumeMounts = append(container.VolumeMounts, corev1.VolumeMount{
-		Name:      volumeName,
-		MountPath: "/newrelic-instrumentation",
-	})
+	if isContainerVolumeMissing(container, volumeName) {
+		container.VolumeMounts = append(container.VolumeMounts, corev1.VolumeMount{
+			Name:      volumeName,
+			MountPath: "/newrelic-instrumentation",
+		})
+	}
 
 	// We just inject Volumes and init containers for the first processed container.
-	if isInitContainerMissing(pod) {
-		pod.Spec.Volumes = append(pod.Spec.Volumes, corev1.Volume{
-			Name: volumeName,
-			VolumeSource: corev1.VolumeSource{
-				EmptyDir: &corev1.EmptyDirVolumeSource{},
-			}})
+	if isInitContainerMissing(pod, javaInitContainerName) {
```

Comment:
If this is not the case, do we assume that the operator already ran once for `java` agent and that `java` is already instrumented?

Reply:
correct.  This is based on the existing code, though it was limited to just the `initContainerName`, which means whichever apm agent ran first would successfully set the container correctly, anything thereafter would be incorrect